server/src/main/java/org/elasticsearch/cluster/metadata/OperatorHandlerMetadata.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.cluster.Diff;
+import org.elasticsearch.cluster.SimpleDiffable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * Metadata class to hold the keys set in operator mode for each operator handler.
+ *
+ * <p>
+ * Since we hold operator metadata state for multiple namespaces, the same handler can appear in
+ * multiple namespaces. See {@link OperatorMetadata} and {@link Metadata}.
+ * </p>
+ */
+public record OperatorHandlerMetadata(String name, Set<String> keys)
```

Comment:
The OperatorHandlerMetadata is stored when we successfully process a new operator cluster state. Here's an example usage from the task that does the final update of the cluster state:

https://github.com/elastic/elasticsearch/pull/86224/files#diff-8ca6818d706ec1720e045ea8a1f4f8e4b5dda47c5610aed2bf8e7d55121d8391R40

specifically here:

https://github.com/elastic/elasticsearch/pull/86224/files#diff-8ca6818d706ec1720e045ea8a1f4f8e4b5dda47c5610aed2bf8e7d55121d8391R128

The only difference is that I renamed the API so instead of:

```
Metadata.Builder metadataBuilder = Metadata.builder(state.metadata()).putOperatorState(operatorMetadataBuilder.build());
```

we'll do:

```
Metadata.Builder metadataBuilder = Metadata.builder(state.metadata()).put(operatorMetadataBuilder.build());
```